smartsim/settings/pbsSettings.py

```diff
@@ -66,9 +69,17 @@ def __init__(
         :param batch_args: overrides for PBS batch arguments, defaults to None
         :type batch_args: dict[str, str], optional
         """
-        self._time: t.Optional[str] = None
-        self._nodes: t.Optional[int] = None
+
         self._ncpus = ncpus
+        self._resources = resources or {}
+
+        resource_nodes = self.resources.get("nodes", None)
```

Comment:
I think this might need to be `resource_nodes = self._resources.get("nodes", None)`

Reply:
I think the assignment on 74 is actually wrong (or rather it doesn't go through the new setter method)